webapp/status_server.py

```diff
@@ -8,20 +8,89 @@
 from flask import session
 from flask import redirect
 from flask import url_for
+import time
+import json
 
 from firebase_admin import auth
 
+import webapp
 
-def status(pb):
+
+def status():
     try:
         user = auth.verify_id_token(session['api_session_token'])
         g.user = user
     except:
         return redirect(url_for("login"))
 
+    nickname, group = get_info(user)
     if request.method == "POST":
         # TODO add task to database
-        return render_template("status.html")
-    else:
+        add_task(group, user)
+        # return render_template("status.html")
+    # else:
         # TODO load user specific data, e.g. tasks, group resources
-        return render_template("status.html")
\ No newline at end of file
+
+    return render_template("status.html", nickname=nickname, group=group)
+
+
+def get_info(user):
+  db = webapp.pb.database()
+  uid = user['user_id']
+
+  nickname = db.child('users').child(uid).child('nickname').get().val()
+  group = db.child('users').child(uid).child('group').get().val()
+
+  return nickname, group
+
+
+def add_task(group, user):
+  if request.method == "POST":
+    db = webapp.pb.database()
+
+    content = request.form.get('task')
+    uid = user['user_id']
+
+    key = db.child('groups').child(group).child(uid).child('tasks').push({
```

Comment:
maybe we want to change this to users -> uid -> tasks?
(and other queries for tasks)